samples/Azure.Management.Storage/Generated/StorageAccountsCreateOperation.cs

```diff
@@ -57,23 +57,21 @@ internal StorageAccountsCreateOperation(ResourceOperationsBase operationsBase, C
         public override ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) => _operation.UpdateStatusAsync(cancellationToken);
 
         /// <inheritdoc />
-        public async override ValueTask<Response<StorageAccount>> WaitForCompletionAsync(CancellationToken cancellationToken = default) => MapResponseType(await _operation.WaitForCompletionAsync(cancellationToken));
+        public override ValueTask<Response<StorageAccount>> WaitForCompletionAsync(CancellationToken cancellationToken = default) => _operation.WaitForCompletionAsync(cancellationToken);
 
         /// <inheritdoc />
-        public async override ValueTask<Response<StorageAccount>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) => MapResponseType(await _operation.WaitForCompletionAsync(pollingInterval, cancellationToken));
+        public override ValueTask<Response<StorageAccount>> WaitForCompletionAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default) => _operation.WaitForCompletionAsync(pollingInterval, cancellationToken);
 
-        private Response<StorageAccount> MapResponseType(Response<StorageAccountData> response) => Response.FromValue(new StorageAccount(_operationBase, response.Value), response.GetRawResponse());
-
-        StorageAccountData IOperationSource<StorageAccountData>.CreateResult(Response response, CancellationToken cancellationToken)
+        StorageAccount IOperationSource<StorageAccount>.CreateResult(Response response, CancellationToken cancellationToken)
         {
             using var document = JsonDocument.Parse(response.ContentStream);
-            return StorageAccountData.DeserializeStorageAccountData(document.RootElement);
+            return new StorageAccount(_operationBase, StorageAccountData.DeserializeStorageAccountData(document.RootElement));
```

Comment:
Background of the PR: in mgmt-plane LRO we wish to wrap the deserialized model with a class that combines both data and operation to the resource.